binaries/cli/src/command/system/start.rs

```diff
@@ -0,0 +1,209 @@
+use crate::command::{Executable, default_tracing};
+use crate::{LOCALHOST, common::connect_to_coordinator};
+use communication_layer_request_reply::TcpRequestReplyConnection;
+use dora_core::topics::DORA_COORDINATOR_PORT_CONTROL_DEFAULT;
+use dora_message::{cli_to_coordinator::ControlRequest, coordinator_to_cli::ControlRequestReply};
+use eyre::{Context, ContextCompat, bail};
+use indicatif::{ProgressBar, ProgressStyle};
+use std::net::{IpAddr, SocketAddr};
+use std::path::PathBuf;
+use std::{fs, path::Path, process::Command, time::Duration};
+
+use super::status::daemon_running;
+
+#[derive(Debug, clap::Args)]
+/// Start coordinator and daemon in local mode (with default config)
+pub struct Start {
+    /// Use a custom configuration
+    #[clap(long, hide = true, value_name = "PATH", value_hint = clap::ValueHint::FilePath)]
+    config: Option<PathBuf>,
+    /// Address of the dora coordinator
+    #[clap(long, value_name = "IP", default_value_t = LOCALHOST)]
+    coordinator_addr: IpAddr,
+    /// Port number of the coordinator control server
+    #[clap(long, value_name = "PORT", default_value_t = DORA_COORDINATOR_PORT_CONTROL_DEFAULT)]
+    coordinator_port: u16,
+}
+
+impl Executable for Start {
+    fn execute(self) -> eyre::Result<()> {
+        default_tracing()?;
+        let coordinator_addr = (self.coordinator_addr, self.coordinator_port).into();
+        start_system(self.config.as_deref(), coordinator_addr)
+    }
+}
+
+#[derive(Debug, Default, serde::Serialize, serde::Deserialize)]
+struct StartConfig {}
+
+fn create_spinner(msg: &str) -> ProgressBar {
+    let pb = ProgressBar::new_spinner();
+    pb.set_style(
+        ProgressStyle::default_spinner()
+            .tick_chars("⣾⣽⣻⢿⡿⣟⣯⣷")
+            .template("{spinner} {msg}")
+            .unwrap(),
+    );
+    pb.set_message(msg.to_string());
+    pb.enable_steady_tick(Duration::from_millis(80));
+    pb
+}
+
+fn print_success(msg: &str) {
+    println!("\x1b[32m✓\x1b[0m {}", msg);
+}
+
+fn print_error(msg: &str) {
+    eprintln!("\x1b[31m✗\x1b[0m {}", msg);
+}
+
+pub fn start_system(config_path: Option<&Path>, coordinator_addr: SocketAddr) -> eyre::Result<()> {
+    let StartConfig {} = parse_config(config_path)?;
+
+    // Start coordinator
+    let spinner = create_spinner("Starting coordinator...");
+    let mut session = match connect_to_coordinator(coordinator_addr) {
+        Ok(session) => {
+            spinner.finish_and_clear();
+            print_success(&format!(
+                "Coordinator already running on {}:{}",
+                coordinator_addr.ip(),
+                coordinator_addr.port()
+            ));
+            session
+        }
+        Err(_) => {
+            if let Err(e) = start_coordinator() {
```

Comment:
This does not take the `coordinator_addr` into account. So it will start the coordinator on the standard port instead of using the port number given as argument.